# ###########################################################################
# #
# # Copyright 2026 Samsung Electronics All Rights Reserved.
# #
# # Licensed under the Apache License, Version 2.0 (the "License");
# # you may not use this file except in compliance with the License.
# # You may obtain a copy of the License at
# #
# # http://www.apache.org/licenses/LICENSE-2.0
# #
# # Unless required by applicable law or agreed to in writing,
# # software distributed under the License is distributed on an
# # "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
# # express or implied. See the License for the specific language
# # governing permissions and limitations under the License.
# #
# # ###########################################################################

From 1559537d64e4324b16ccf6ea9eaaa572653f6f6e Mon Sep 17 00:00:00 2001
From: anjiahao <anjiahao@xiaomi.com>
Date: Tue, 22 Feb 2022 11:21:24 +0800
Subject: [PATCH 10/21] [test]:modified ltp,rwlock need init

VELAPLATFO-806

Signed-off-by: anjiahao <anjiahao@xiaomi.com>
Change-Id: I446226f2f33cb146e8f39935128204f6a918720d
---
 .../interfaces/pthread_rwlock_trywrlock/speculative/3-1.c       | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/testcases/open_posix_testsuite/conformance/interfaces/pthread_rwlock_trywrlock/speculative/3-1.c b/testcases/open_posix_testsuite/conformance/interfaces/pthread_rwlock_trywrlock/speculative/3-1.c
index cc0d3a296..da1baa1ef 100644
--- a/testcases/open_posix_testsuite/conformance/interfaces/pthread_rwlock_trywrlock/speculative/3-1.c
+++ b/testcases/open_posix_testsuite/conformance/interfaces/pthread_rwlock_trywrlock/speculative/3-1.c
@@ -23,7 +23,7 @@
 int main(void)
 {
 
-	static pthread_rwlock_t rwlock;
+	static pthread_rwlock_t rwlock = PTHREAD_RWLOCK_INITIALIZER;
 	int rc;
 
 	/* Call without initializing rwlock */
-- 
2.40.1

